linkerd/app/outbound/src/http/detect.rs

```diff
@@ -6,7 +6,9 @@ use linkerd_app_core::{
 use tracing::debug_span;
 
 impl<T> Outbound<T> {
-    pub fn push_detect_http<TSvc, H, HSvc, I>(
+    /// Pushes a stack that detects HTTP, unless the service profile indicates
+    /// that this target uses an opaque protocol.
+    pub fn push_detect_with_skip<TSvc, H, HSvc, I>(
```

Comment:
nit: should probably be called `push_detect_http_with_skip` to be clear that's what we're detecting.

Also, I don't really think this should be separate from the main `push_detect_http`, ideally... I.e. could we have a marker param like `SkipHttpDetect` and require that targets of this stack impl `Param<SkipHttpDetect>`? this would help to decouple this from `tcp::Logical` and make it so that we don't need separate  stack builders (and duplicate type constraints)

Reply:
> Also, I don't really think this should be separate from the main `push_detect_http`, ideally... I.e. could we have a marker param like `SkipHttpDetect` and require that targets of this stack impl `Param<SkipHttpDetect>`? this would help to decouple this from `tcp::Logical` and make it so that we don't need separate stack builders (and duplicate type constraints)

Yeah, I didn't like that we ended up with two functions here, but I couldn't figure out a nice way to do it...the `Param` idea is a good one!

Reply:
update, got rid of this in 8f9d9efdf1813afe370187c8c6990b414c8e93d3